doc/nrf/gs_modifying.rst

```diff
@@ -95,15 +122,80 @@ Typically, the :file:`CMakeLists.txt` files include at least the :file:`main.c`
+* Partition manager configuration files that focus on memory layout.
+
+Software-related configuration
+==============================
+
+The software-related configuration comes from Kconfig files (called simply :file:`Kconfig`).
+There are several layers that provide the configuration options:
+
+* The lowest layer consists of defaults provided in Kconfig files of all system components, libraries, and application itself.
+* The defaults are overlayed by the options specified by the selected board in its :file:`*_defconfig` file and its :file:`Kconfig.defconfig` file (see :ref:`zephyr:default_board_configuration` in the Zephyr documentation for more information).
+* The configuration is additionally overlayed by the :file:`prj.conf` file provided by the application.
+
+After merging the configuration sets imported from these configuration files, the build system creates a single :file:`.config` file that describes the entire software configuration of the system.
+This is done once and configuration comes solely from the :file:`.config` file until the project is reevaluated.
+
+.. note::
+   You can edit the :file:`.config` file and rebuild the application to check how the configuration changes.
+   The changes will be lost when the application project is reevaluated.
+   To store the configuration permanently you need to identify which file should provide desired option and edit it directly.
+   In most cases this will be your application :file:`prj.conf` file.
 
 For detailed information about configuration options, see :ref:`zephyr:application-kconfig` in the Zephyr documentation.
 
+Hardware-related configuration
+==============================
+
+The hardware-related configuration comes from :file:`dts` files.
+There are several layers that provide the hardware definitions:
+
+* The lowest layer consists of defaults provided in :file:`dtsi` files for the given SoC.
+* The :file:`dtsi` files are included by selected board in its :file:`dts` file, which provides defaults of hardware configuration for the entire board.
+* The hardware-related configuration is further overlayed by the :file:`app.overlay` file, which is provided by the application.
+
+After merging the configuration sets imported from these configuration files, the build system creates a single :file:`zephyr.dts` file that describes the entire hardware-related configuration of the system.
+This process also creates a set of header files that provide defines, which can be used to refer hardware configuration from the source code.
+The hardware-related configuration does not change until the project is reevaluated.
+
+.. note::
+   The files resulting from DTS evaluation should not be notified.
+   To store the configuration parmanantly, you need to identify the file that contributes to the specific part of DTS tree and edit it directly.
+   In most cases, this is the :file:`app.overlay` file of your application.
```

Comment:
@mbolivar-nordic would you please take a close look at this part.

I would describe devicetree as `Hardware-related configuration`, even though a small portion of devicetree (like the `chosen` node part and io-pin overlay) can be seen as configuration.

But you are the master on how to describe this.

Reply:
DTS gives more configuration options that affect application. That can be order in which nodes are defined. Channel selection, speeds, etc.
We should also look from the point of view of the integrator that modifies the application as a whole (meaning they want to extend the behavior to new HW). DTS configures HW. Since application can change it using overlay I believe this must be mentioned.

Reply:
I know what devicetree is.
And I don't think this section is precise enough, and thus might mislead the reader into thinking devicetree is something else.

Btw. why are we trying to repeat devicetree documentation when we already have this good description in nRF Connect SDK docs:
http://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/guides/dts/intro.html

Also note from that page:
> There are two types of devicetree input files: *devicetree sources* and *devicetree bindings*.

Reply:
I don't try to repeat it. If this page is a central hub about how to configure application we need some introduction about various points where application configuration takes place.

Reply:
Added the link to the DTS intro in Zephyr in the first sentence, as suggested by @SebastianBoe . The rest was not touched, because of what @pdunaj wrote.